### DISCUSSION TITLE
v2ray api 流量统计只能统计所有inbound

### DISCUSSION THREAD

**Discussion by nkeonkeo** [v2fly/v2ray-core#510] [Ideas]
api能否添加分inbound统计?

**Comment by ghost** [comment]
目前是支持分 inbound 统计的，不知道你怎么判断的问题

**Reply by ghost** [reply]
```
> v2ctl api --server=127.0.0.1:15490 StatsService.QueryStats 'pattern: "" reset: false'

stat: <
  name: "inbound>>>QV2RAY_API_INBOUND>>>traffic>>>downlink"
  value: 9698165
>
stat: <
  name: "inbound>>>socks_IN>>>traffic>>>uplink"
>
stat: <
  name: "outbound>>>dns-out>>>traffic>>>uplink"
>
stat: <
  name: "outbound>>>PROXY>>>traffic>>>downlink"
  value: 471
>
stat: <
  name: "outbound>>>DIRECT>>>traffic>>>downlink"
  value: 31
>
stat: <
  name: "inbound>>>QV2RAY_API_INBOUND>>>traffic>>>uplink"
  value: 15334859
>
stat: <
  name: "inbound>>>tproxy_IN>>>traffic>>>downlink"
  value: 31
>
stat: <
  name: "outbound>>>PROXY>>>traffic>>>uplink"
  value: 634
>
stat: <
  name: "outbound>>>BLACKHOLE>>>traffic>>>downlink"
>
stat: <
  name: "inbound>>>socks_IN>>>traffic>>>downlink"
>
stat: <
  name: "inbound>>>tproxy_IN>>>traffic>>>uplink"
  value: 297
>
stat: <
  name: "inbound>>>tproxy_IN_V6>>>traffic>>>downlink"
>
stat: <
  name: "outbound>>>DIRECT>>>traffic>>>uplink"
>
stat: <
  name: "outbound>>>BLACKHOLE>>>traffic>>>uplink"
>
stat: <
  name: "outbound>>>dns-out>>>traffic>>>downlink"
>
stat: <
  name: "inbound>>>http_IN>>>traffic>>>uplink"
>
stat: <
  name: "inbound>>>http_IN>>>traffic>>>downlink"
>
stat: <
  name: "inbound>>>tproxy_IN_V6>>>traffic>>>uplink"
>
```

**Comment by lp123sun** [comment]
------------Inbound----------
inbound:in-0->up    76MB
inbound:in-0->down  6.6GB
inbound:api->up     173B
inbound:api->down   24B
SUM->up:            76MB
SUM->down:          6.6GB
SUM->TOTAL:         6.6GB
-----------------------------

-------------User------------
user:1->up      52MB
user:1->down    172MB
user:2->up    12MB
user:2->down  6.1GB
user:3->up    4.5MB
user:3->down  151MB
SUM->up:                        68MB
SUM->down:                      6.4GB
SUM->TOTAL:                     6.5GB
-----------------------------

可以按用户统计的

**Reply by lp123sun** [reply]
有User统计了，再在inbound分别统计意义在哪里呢

**Reply by GeorgeRudd** [reply]
> ## ------------Inbound----------
> inbound:in-0->up 76MB
> inbound:in-0->down 6.6GB
> inbound:api->up 173B
> inbound:api->down 24B
> SUM->up: 76MB
> SUM->down: 6.6GB
> SUM->TOTAL: 6.6GB
> ## -------------User------------
> user:1->up 52MB
> user:1->down 172MB
> user:2->up 12MB
> user:2->down 6.1GB
> user:3->up 4.5MB
> user:3->down 151MB
> SUM->up: 68MB
> SUM->down: 6.4GB
> SUM->TOTAL: 6.5GB
> 可以按用户统计的

你好 可以展示下你的配置吗 为啥我的是user3在最上面 user1在下面

**Comment by lp123sun** [comment]
只是每次重启v2，流量统计都会清零

**Reply by zhi1ong** [reply]
crontab 定期轮转统计结果落盘就可以了。。
